contracts/src/LightClient.sol

```diff
@@ -84,12 +87,19 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     /// @notice a flag that indicates when a permissioned provrer is needed
     bool public permissionedProverEnabled;
 
-    /// @notice an array to store the L1 Block Heights where the finalizedState was updated
-    uint256[] public stateUpdateBlockNumbers;
+    ///@notice Max number of seconds worth of state commitments to record based on this block
+    /// timestamp
+    uint32 public maxStateHistoryDuration;
```

Comment:
nit: I think instead of "max" we should call it "keep" (or something like this) because it's the desired amount of history we will keep. Similar for `maxHistorySeconds`, `keepHistorySeconds` IMO expresses more clearly what the variable is supposed to do.

Reply:
That's fair, I renamed it to "stateHistoryRetentionPeriod" https://github.com/EspressoSystems/espresso-sequencer/commit/fac2e76df421bf0d0e584de82297736f7dbfbc89

Reply:
Nice